magento/magento-cloud-patches
# MCLOUD-5666: Assign every patch to modules it changes

### Description
Assign patches for PayPal, Braintree and Authorize.net to appropriate Magento modules

### Fixed Issues (if relevant)
1. https://jira.corp.magento.com/browse/MCLOUD-5666

### Manual testing scenarios

There are 4 patches were unassigned from magento/magento2-base and assigned to modules they change:
A  MAGETWO-53941__fix_enterprise_paypal_codes__100.1.6.patch for module-paypal
B. MC-31387__fix_paypal_issue_with_region__100.3.4.patch for module-paypal
C. MAGETWO-98129__transaction_field_is_removed_by_authorize_net__100.2.0.patch for module-autorizenet
D. MAGETWO-53941__fix_enterprise_braintree_code__100.1.6.patch for module-braintree

1. Check applying patches on next Magento versions:
- 2.1.7 - none of the patches applied
- 2.1.8 - patch A and D applied
- 2.1.18 - patch A and D applied
- 2.2.0 - only patch C applied
- 2.2.7 - only patch C applied
- 2.2.8 - none of the patches applied
- 2.3.4 - Patch B applied

2. Uninstall modules magento/module-paypal, magento/module-braintree and magento/module-autorizenet. Check on next Magento versions that none of those patches applied and deployment was finished without errors:
2.1.8, 2.2.0, 2.3.4


### Contribution checklist

releasenotes
Updated patch assignment for payment method patches to target the payment modules instead of the Magento base package (`magento/magento2-base`) so that the payment patches are applied only if the payment modules exist.
